sql_assesment.txt

```diff
@@ -0,0 +1,1786 @@
+
+
+select sname, cname from customers, salespeople where salespeople.snum= customers.snum and sname<cname;
+
++---------+----------+
+| sname   | cname    |
++---------+----------+
+| AxelRod | Giovanni |
+| Motika  | Periera  |
++---------+----------+
+
+
+
+59) Write a query that produces the names and ratings of all customers who have average orders. 
+
+SELECT C.CNUM,CNAME, RATING, COUNT(O.CNUM) AS AVERAGE_ORDERS FROM CUSTOMERS C JOIN ORDERS O ON C.CNUM=O.CNUM GROUP BY O.CNUM HAVING COUNT(O.CNUM)=(SELECT ROUND(AVG(TOTAL)) AS AVG_VALUE FROM (SELECT COUNT(CNUM) AS TOTAL FROM ORDERS GROUP BY CNUM) AS TABLE1);
+
++------+----------+--------+----------------+
+| CNUM | CNAME    | RATING | AVERAGE_ORDERS |
++------+----------+--------+----------------+
+| 2001 | Hoffman  |    100 |              1 |
+| 2007 | Periera  |    100 |              1 |
+| 2003 | Liu      |    200 |              1 |
+| 2002 | Giovanni |    200 |              1 |
++------+----------+--------+----------------+
+
+
+
+
+60) Find the SUM of all Amounts from the orders table. 
+
+
+select sum(amt) from orders;
+
++----------+
+| sum(amt) |
++----------+
+|  26658.4 |
++----------+
+
+
+
+61) Write a SELECT command that produces the order number, amount, and the date from rows in the order table. 
+
+select onum, amt ,odate from orders;
+
++------+---------+------------+
+| onum | amt     | odate      |
++------+---------+------------+
+| 3001 |   18.69 | 1990-03-10 |
+| 3003 |  767.19 | 1990-03-10 |
+| 3002 |  1900.1 | 1990-03-10 |
+| 3005 | 5160.45 | 1990-03-10 |
+| 3006 | 1098.16 | 1990-03-10 |
+| 3009 | 1713.23 | 1990-04-10 |
+| 3007 |   75.75 | 1990-04-10 |
+| 3008 |    4723 | 1990-05-10 |
+| 3010 | 1309.95 | 1990-06-10 |
+| 3011 | 9891.88 | 1990-06-10 |
++------+---------+------------+
+
+
+
+
+62) Count the number of non NULL rating fields in the Customers table (including repeats). 
+
+
+select count(rating) from customers where rating is not null;
+
++---------------+
+| count(rating) |
++---------------+
+|             7 |
++---------------+
+
+
+
+63) Write a query that gives the names of both the salesperson and the customer for each order after the order number.  
+
+
+select onum, sname ,cname from salespeople, orders , customers  where salespeople.snum= customers.snum  and orders.cnum= customers.cnum;
+
++------+---------+----------+
+| onum | sname   | cname    |
++------+---------+----------+
+| 3003 | Peel    | Hoffman  |
+| 3011 | Peel    | Ciemens  |
+| 3008 | Peel    | Ciemens  |
+| 3005 | Serres  | Liu      |
+| 3010 | Serres  | Grass    |
+| 3007 | Serres  | Grass    |
+| 3009 | AxelRod | Giovanni |
+| 3002 | Motika  | Periera  |
+| 3006 | Rofkin  | Cisneros |
+| 3001 | Rofkin  | Cisneros |
++------+---------+----------+
+
+
+
+
+64) List the commissions of all salespeople servicing customers in London.
+
+
+select sname, comm from salespeople where city="London";
+
```

Comment:
Incorrect. The customers' city must be "London" not that of Salespeople. Please correct it.